### DISCUSSION TITLE
Unable to Connect MS SQL

### DISCUSSION THREAD

**Discussion by luisdgeneMT** [Budibase/budibase#11204] [Help] [answered]
When editing and using my apps, I am no longer able to connect to my **MS SQL** data sources, whether they are existing or when attempting to connect to a new one. I have tried with two different servers and multiple databases and I get this error:

**When attempting to create a new connection:**
Error creating datasource: Unable to connect: No such case in exhaustive switch: [object Object]

**When loading an existing table in the Data section:**
There was a problem loading your grid
No such case in exhaustive switch: [object Object]

This also reflects in the corresponding published apps.

This started sometime in the last 12 hours. I tried upgrading from 2.8.5 to 2.8.9, **oddly it worked on an app on version 2.7.36** but not in an app in 2.7.24. Out of 10+ apps I tried, it only worked on that single one.

Running Brave Version 1.52.130 Chromium: 114.0.5735.198 (Official Build) (64-bit)

**Comment by melohagan** [comment]
Hey @luisdgeneMT 

Could you please provide a video recording, or some screenshots at the least? It will help us diagnose the problem faster.

**Reply by luisdgeneMT** [reply]
Published App 2.8.9:
![image](https://github.com/Budibase/budibase/assets/139163046/5b319634-2223-4c36-a63f-2549686295e2)
(Edit: Corrected screenshot)

![image](https://github.com/Budibase/budibase/assets/139163046/c54393c9-c035-4960-b16b-ec051c52605c)

![image](https://github.com/Budibase/budibase/assets/139163046/e024bf11-48cb-403e-98c2-a4d972b4e5df)

While troubleshooting I have found that the apps that work do not fetch tables that have been created or altered in Budibase.
However, when attempting to create a new connection with a blank app, it will not let me connect - even to databases that I have never fetched before.

In the -now 2- apps that I have found that work, even if I attempt to create a **new** connection, I get the same error, but the existing connections keep working.

**Reply by adrinr** [reply, marked as answer]
I can confirm the error. This has been introduced in 2.8.9. Fix in this PR: https://github.com/Budibase/budibase/pull/11205

**Reply by adrinr** [reply]
Version 2.8.10 is out with the fix. Could you confirm, @luisdgeneMT?

**Reply by luisdgeneMT** [reply]
I can confirm all connections have been restored and work as intended! Thanks for the great work!
